src/core/ddsi/src/q_receive.c

```diff
@@ -2891,6 +2891,9 @@ static bool do_packet (struct thread_state1 * const ts1, struct q_globals *gv, d
   {
     return false;
   }
+
```

Comment:
This is good — but ``DDSRT_STATIC_ASSERTION`` would be even better: that turns it into a compile-time check, so that you can't build it if the two aren't the same.